File: src/test/java/example/example/util/ReportUtil.java
package example.example.util;

import java.util.Base64;

import com.relevantcodes.extentreports.LogStatus;

import example.example.context.WebDriverContext;
import example.example.report.ExtentReportManager;

public class ReportUtil {

	public static void addScreenShot(String message) {
		byte[] screenshotBytes = WebDriverContext.getDriver().screenshot();
		String base64Image = "data:image/png;base64," + Base64.getEncoder().encodeToString(screenshotBytes);
		ExtentReportManager.getCurrentTest().log(LogStatus.INFO, message,
				ExtentReportManager.getCurrentTest().addBase64ScreenShot(base64Image));
	}

	public static void addScreenShot(LogStatus status, String message) {
		byte[] screenshotBytes = WebDriverContext.getDriver().screenshot();
		String base64Image = "data:image/png;base64," + Base64.getEncoder().encodeToString(screenshotBytes);
		ExtentReportManager.getCurrentTest().log(status, message,
				ExtentReportManager.getCurrentTest().addBase64ScreenShot(base64Image));
	}

	public static void logMessage(String message, String details) {
		ExtentReportManager.getCurrentTest().log(LogStatus.INFO, message, details);
	}

	public static void logMessage(LogStatus status, String message, String details) {
		ExtentReportManager.getCurrentTest().log(status, message, details);
	}
}

File: src/test/python/example/example/util/report_util.py
from selenium.webdriver.remote.webdriver import WebDriver

from example.example.context.web_driver_context import get_driver
from example.example.util.logger_util import log


def add_screenshot(message: str, status: str = "INFO") -> str:
    try:
        driver: WebDriver = get_driver()
        screenshot_base64 = driver.get_screenshot_as_base64()
        base64_image = f"data:image/png;base64,{screenshot_base64}"
        log(f"[{status}] {message}")
        return base64_image
    except Exception as e:
        log(f"Failed to capture screenshot: {e}")
        return ""


def log_message(message: str, details: str, status: str = "INFO") -> None:
    log(f"[{status}] {message} - {details}")
